api/app/mailers/notify_team_leader_mailer.rb

```diff
@@ -0,0 +1,9 @@
+class NotifyTeamLeaderMailer < ApplicationMailer
+  def notify_team_leader(task_id, user_id)
+    @task = Task.find_by(id: task_id)
+    @owner = User.find_by(id: user_id)
```

Comment:
This definitely works and is clear. A bit more idiomatic in Ruby is to use the ActiveRecord asspciation methods. You could use `@owner = task.user` instead of `@owner = User.find_by(id: user_id)`. Note that it still does the same number of database calls... its just a bit easier to read.

You could then change the method signature to `def notify_team_leader(task_id)`

Reply:
Definitely makes sense!

So when you declare a variable in ruby you don't need to always call the @ sign in front of it? I guess my question is better stated as it doesn't need to be `@owner = @task.user`?